rubin_scheduler/scheduler/detailers/detailer.py

```diff
@@ -202,6 +203,55 @@ def __call__(self, observation_array, conditions):
         return observation_array
 
 
+class TruncatePreTwiDetailer(BaseDetailer):
+    """Truncate an array of observations so they fit before
+    morning twilight starts.
+
+    Parameters
+    ----------
+    pad : `float`
+        The pad to give before start of morning twilight.
+        Default 5 (minutes).
+    filter_change_time : `float`
+        Estimate of filter change time
+    visit_overhead : `float`
+        Estimate of visit overhead (read, slew, settle).
+        Default 2.5 (seconds).
+    twilight_cut_to : `str`
+        Cut to sun altitude of -18 degrees or -12 degrees.
+        Valid values of "n18" or "n12".
+    """
+
+    def __init__(self, pad=5.0, filter_change_time=120.0, visit_overhead=2.5, twilight_cut_to="n18"):
```

Comment:
We're slewing between DDF visits within a night now, right? Isn't 2.5 too short? (this is already shorter than the readout time).

Reply:
Sure, we can bump that up. With the 5-minute pad default it shouldn't end up mattering in most situations.